samples/cellular/modem_shell/src/fota/fota_shell.c

```diff
@@ -33,8 +33,7 @@ static int cmd_fota_download(const struct shell *shell, size_t argc,
 	} else if (strcmp(argv[1], "au") == 0) {
 		fota_server = fota_server_au;
 	} else {
-		mosh_error("FOTA: Unknown server: %s", argv[1]);
-		return -EINVAL;
+		fota_server = argv[1];
```

Comment:
This does not seem to work. `fota download eu ncs_ci_mfw_nrf91x1_update_from_2.0.1-FOTA-TEST_to_2.0.1.bin` works, but `fota download https://nrf-test-eu.s3.amazonaws.com/ncs_ci_mfw_nrf91x1_update_from_2.0.1-FOTA-TEST_to_2.0.1.bin foo` does not. The extra parameter after the URI is currently mandatory, see my other comment.

Reply:
I've removed support for giving the URI in this PR, host and file must be separate. It should work with `fota download https://nrf-test-eu.s3.amazonaws.com ncs_ci_mfw_nrf91x1_update_from_2.0.1-FOTA-TEST_to_2.0.1.bin`

Reply:
The implementation still seems be there, see line 36. I think you should restore the previous error handling.

Also the help text needs to be restored, see lines 58-60.

Reply:
The idea is that if argv[1] is not recognized as one of the servers (eu, au, ..) it will be used directly as the server address, though the file path must be given in a separate argument. I found this helpful as you can then use it to download from other servers than the ones that are provided by giving the url.

Reply:
Ok, now I see it. I read the help text too quickly. 😃